File: java-concert/src/main/java/org/example/javaconcert/concert/domain/ConcertRepository.java
package org.example.javaconcert.concert.domain;

import java.util.List;

import org.example.javaconcert.concert.infrastructure.ConcertJpaRepository;
import org.example.javaconcert.concert.infrastructure.entity.ConcertEntity;
import org.springframework.stereotype.Repository;

import lombok.RequiredArgsConstructor;

@Repository
@RequiredArgsConstructor
public class ConcertRepository {

	private final ConcertJpaRepository concertJpaRepository;

	public List<Concert> findAll() {
		return concertJpaRepository.findAll().stream()
			.map(ConcertEntity::toDomain)
			.toList();
	}

	public Long save(Concert concert) {
		ConcertEntity entity = concertJpaRepository.save(ConcertEntity.toEntity(concert));
		return entity.getId();
	}
}

File: concert/src/main/kotlin/prac/concert/concert/domain/ConcertRepository.kt
package prac.concert.concert.domain

import org.springframework.stereotype.Component
import prac.concert.concert.infrastructure.ConcertJpaRepository
import prac.concert.concert.infrastructure.entity.ConcertEntity

@Component
class ConcertRepository(
    private val concertJpaRepository: ConcertJpaRepository
) {

    fun findAll(): List<Concert>{
        return concertJpaRepository.findAll()
            .map { entity -> entity.toDomain() }
    }

    fun findById(id : Long) : Concert{
        return concertJpaRepository.findById(id)
            .orElseThrow{ IllegalArgumentException("id가 없습니다.") }
            .toDomain()
    }

    fun save(concert: Concert): Long{
        val entity = concertJpaRepository.save(ConcertEntity.toEntity(concert))
        return entity.id;
    }
}
